### PR TITLE
Allow opening a map view in OpenStreetMap

### PR DESCRIPTION
**Pull request by andybalaam**
Builds on https://github.com/matrix-org/matrix-react-sdk/pull/7424 .  (I can't base this off that because the branch is on my repo, but in future I will push branches to this repo to make it easier.)  This adds one commit: 07b3d4c723

open-in-openstreetmap

---
Here's what your changelog entry will look like:

## ✨ Features
 * Allow opening a map view in OpenStreetMap ([\#7428](https://github.com/matrix-org/matrix-react-sdk/pull/7428)). Contributed by @andybalaam.

Preview: https://61c1f67ba7ed89213999bd70--matrix-react-sdk.netlify.app
⚠️ Do you trust the author of this PR? Maybe this build will steal your keys or give you malware. Exercise caution. Use test accounts.

### CODE REVIEW

File: src/components/views/context_menus/MessageContextMenu.tsx [hunk 1]
```diff
@@ -133,9 +142,13 @@ export default class MessageContextMenu extends React.Component<IProps, IState>
...
     }
 
+    private canOpenInMapSite(mxEvent: MatrixEvent): boolean {
+        return isLocationEvent(mxEvent);
+    }
+
     private canEndPoll(mxEvent: MatrixEvent): boolean {
         return (
-            mxEvent.getType() === POLL_START_EVENT_TYPE.name &&
+            POLL_START_EVENT_TYPE.matches(mxEvent.getType()) &&
```
Comment (turt2live) [inline src/components/views/context_menus/MessageContextMenu.tsx:151]: <3 thank you

File: src/components/views/messages/MLocationBody.tsx [hunk 2]
```diff
@@ -147,3 +148,29 @@ export function parseGeoUri(uri: string): GeolocationCoordinates {
         speed: undefined,
     };
 }
+
+function makeLink(coords: GeolocationCoordinates): string {
+    return (
+        "https://www.openstreetmap.org/" +
```
Comment (turt2live) [inline src/components/views/messages/MLocationBody.tsx:154]: can we make this a config.json variable to support self-hosted setups? Not critical for this PR, but something we should do before launch I think.

**Review by turt2live — APPROVED** [review]
Thanks!

File: src/components/views/messages/MLocationBody.tsx (hunk 2 above)
Comment (andybalaam) [reply src/components/views/messages/MLocationBody.tsx:154]: Thanks, I'll add a story.  I guess we'd need to configure the text of the button as well

Comment (andybalaam) [reply src/components/views/messages/MLocationBody.tsx:154]: Story created in internal tracker.

**Review by gaelledel — COMMENTED** [review]
All good!

File: src/components/views/messages/MLocationBody.tsx (hunk 2 above)
Comment (turt2live) [reply src/components/views/messages/MLocationBody.tsx:154]: Button text is probably less of a priority - will copy to internal ticket.

**Pull request merged**